### DISCUSSION TITLE
Which source paths are checked by detekt by default?

### DISCUSSION THREAD

**Discussion by kawpii** [detekt/detekt#3467] [Q&A] [answered]
I am using detekt gradle plugin, I wonder which source paths are checked by detekt by default?

Is it only ?

- `main/java/`
- `main/kotlin/`
- `test/java/`
- `test/kotlin/`

what about `.kts` files?

Do I need a custom detekt task to include `.kts` and test source files as described [here](https://detekt.github.io/detekt/gradle.html#kotlin-dsl-6)?

**Comment by cortinico** [comment, marked as answer]
> I am using detekt gradle plugin, I wonder which source paths are checked by detekt by default?

The answer depends on which detekt task you're invoking. I will answer assuming you're invoking the `detetk` plain task and not providing further configuration. Then the task will check:
* Inside the `src/main/java` and `src/main/kotlin` folders ([see here](https://github.com/detekt/detekt/blob/899ad5ba588b2774b9993f93c265ace429dbfb51/detekt-gradle-plugin/src/main/kotlin/io/gitlab/arturbosch/detekt/extensions/DetektExtension.kt#L68-L69))
* Both `.kt` and `.kts` files ([see here](https://github.com/detekt/detekt/blob/899ad5ba588b2774b9993f93c265ace429dbfb51/detekt-gradle-plugin/src/main/kotlin/io/gitlab/arturbosch/detekt/DetektPlugin.kt#L97))

**Reply by kawpii** [reply]
thanks for reply! Yes I am talking about `detekt` task, is `build.gradle.kts` inside modules (ex: app/ ) also checked? because they are not in `src/main/java` or `src/main/kotlin`

Is not [line](https://github.com/detekt/detekt/blob/899ad5ba588b2774b9993f93c265ace429dbfb51/detekt-gradle-plugin/src/main/kotlin/io/gitlab/arturbosch/detekt/DetektPlugin.kt#L97) check all *.kt, *.kts files regardless of the path? or is it local to the src set directories?

**Reply by cortinico** [reply]
> is build.gradle.kts inside modules (ex: app/ ) also checked? because they are not in src/main/java or src/main/kotlin

Nope, it's not inspected by default as is not in the two folders I mentioned.

> Is not line check all *.kt, *.kts files regardless of the path? or is it local to the src set directories?

Correct, the `.kt` and `.kts` files are searched only in the `src/main/java` and `src/main/kotlin` directories.

**Reply by kawpii** [reply]
> Nope, it's not inspected by default as is not in the two folders I mentioned.

So that means we need to define [custom detekt task](https://detekt.github.io/detekt/gradle.html#defining-custom-detekt-task) in order to check build.gradle.kts That is annoying which there was config option to scan build.gradle.kts files. Should I open a feature request for this?

**Reply by cortinico** [reply]
> So that means we need to define custom detekt task in order to check build.gradle.kts

Not necessarily. You can configure Detekt's input using the `detekt{}` block in your gradle file ([see here](https://detekt.github.io/detekt/gradle.html)):

```
detekt {
    input = files(     // The directories where detekt looks for source files. Defaults to `files("src/main/java", "src/main/kotlin")`.
        "src/main/kotlin",
        "gensrc/main/kotlin"
    )
    ...
}
```

**Reply by Marcono1234** [reply]
It looks like since #3649 the test sources will be checked as well.
